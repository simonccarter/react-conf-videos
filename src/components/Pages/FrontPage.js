import React from 'react'
import { connect } from 'react-redux'
import { compose, pure } from 'recompose'
import cn from 'classnames'
import { List } from 'components/List'

import styles from './FrontPage.scss'

const FrontPageInner = props => (
  <main className={cn(styles.FrontPage, { [`${styles.isActive}`]: props.isActive })} >
    <div className={styles.header} >
      <div className={cn(styles.text, { [`${styles.isActive}`]: props.isActive })} >
        <h1> React.js Videos</h1>
        <p> Search React.js conference videos. </p>
        <p className={styles.contribute}> Contribute <a href="https://github.com/simonccarter/react-conf-videos">here</a>.</p>
      </div>
    </div>
    <List />
  </main>
)

const mapStateToProps = state => ({
  isActive: state.frontPage.isActive
})

const FrontPage = compose(
  connect(mapStateToProps),
  pure
)(FrontPageInner)

export default FrontPage
