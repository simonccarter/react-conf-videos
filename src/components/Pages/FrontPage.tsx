import * as React from 'react'
import * as cn from 'classnames'
import { connect } from 'react-redux'
import { compose, pure } from 'recompose'

import { List } from  'components/List'

import styles from './FrontPage.scss'

type Props = { isActive: boolean }

const FrontPageInner: React.SFC<Props> = props => (
  <main className={cn(styles.FrontPage, { [`${styles.isActive}`]: props.isActive })} >
    <div className={styles.header} >
      <div className={cn(styles.text, { [`${styles.isActive}`]: props.isActive })} >
        <h1> React.js Videos</h1>
        <p> Search React.js conference videos. </p>
        <p className={styles.contribute}> Contribute <a href="https://github.com/simonccarter/react-conf-videos">here</a>.</p>
      </div>
    </div>
    <List />
  </main>
)

const mapStateToProps = (state: any) => ({
  isActive: state.frontPage.isActive
})

const FrontPage = compose<Props, {}>(
  connect(mapStateToProps),
  pure
)(FrontPageInner)

export default FrontPage
