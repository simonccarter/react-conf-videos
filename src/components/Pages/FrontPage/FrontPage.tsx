import * as React from 'react'
import { connect } from 'react-redux'
import { compose, pure } from 'recompose'
import { List, ResultDetails } from 'components'

import { 
  SearchInput,
  InnerLayoutContainer
} from  'components'

import { IndexedConferences } from '../../../domain'

type ReduxProps = { 
  conferences: IndexedConferences
}
type Props = ReduxProps

export const FrontPageInner: React.SFC<Props> = props => (
  <InnerLayoutContainer>
    <SearchInput />
    <ResultDetails conferences={props.conferences} />
    <List conferences={props.conferences} />
  </InnerLayoutContainer>
)

const mapStateToProps = ({frontPage: {conferences}}: any) => ({
  conferences
})

const FrontPage = compose<Props, {}>(
  connect(mapStateToProps),
  pure
)(FrontPageInner)

export default FrontPage
