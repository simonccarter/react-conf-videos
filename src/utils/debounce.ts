export default (func: (...idx: unknown[]) => void, delay = 300) => {
  let timer: NodeJS.Timeout;

  return (...args: unknown[]) => {
    clearTimeout(timer);
    timer = setTimeout(() => func(...args), delay);
  };
};
