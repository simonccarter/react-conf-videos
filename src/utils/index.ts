export * from './test'
export * from './search'
export * from './url'