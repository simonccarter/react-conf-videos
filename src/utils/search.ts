const isFilterEmpty = (filterValue: string = '') => filterValue.trim() === ''

export { isFilterEmpty }