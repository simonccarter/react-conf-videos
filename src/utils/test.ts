import { createMockStore } from 'redux-test-utils'
import { MiddlewareAPI } from 'redux'
import { mount, shallow } from 'enzyme'
import { ReactElement } from 'react'

import { Conference } from '../domain'

/** epics **/

// fails test of error cb of subcribe is called
export const onError = (done: any) => (error: Error) => done(false)

// mock empty store object
export const mockStore = () => { return <MiddlewareAPI<void>>{} }

/** Components **/

const isStore = (store: any) => store.hasOwnProperty('isActionTypeDispatched')

// shallow mount component with a store in context
export const shallowWithStore = <T>(storeP: T, component: ReactElement<any>) => {
  const store = isStore(storeP) ? storeP : createMockStore(storeP)
  return shallow<any, any>(component, { context: { store } })
}

export const mountWithStore = <T>(storeP: T, component: ReactElement<any>) => {
  const store = isStore(storeP) ? storeP : createMockStore(storeP)
  return mount<any, any>(component, { context: { store } })
}

/** Mock data */
export const mockConference = (): Conference => ({
  date: 'XX/YY/ZZZ',
  title: 'fake conf title',
  website: 'fake url',
  playlist: 'day 1',
  videos: ['aaa', 'bbb', 'ccc']
})