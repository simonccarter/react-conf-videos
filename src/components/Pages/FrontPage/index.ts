import FrontPage from './FrontPage'

export { FrontPage }
