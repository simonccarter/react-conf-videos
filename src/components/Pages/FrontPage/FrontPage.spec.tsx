import * as React from 'react'
import { shallow } from 'enzyme'
import { shallowWithStore } from 'utils'
import FrontPage, { FrontPageInner } from './FrontPage'
import toJson from 'enzyme-to-json'

import { mockConference } from 'utils'

describe('FrontPage', () => {
  describe.skip('FrontPageInner', () => {

    const shallowComp = (isActive: boolean) => shallow<any>(
      <FrontPageInner
        conferences={{'x': mockConference()}}
        filterValue=''
      />
    )

    it('should render', () => {
      // arrange
      // act
      const wrapper = shallowComp(true)

      // expect
      expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('should not have the isActive classes if isActive is false', () => {
      // arrange
      // act
      const wrapper = shallowComp(false)

      // assert
      expect(wrapper.find('.frontPage.isActive').length).toBe(0)
      expect(wrapper.find('.text.isActive').length).toBe(0)
    })
  })

  describe('FrontPage', () => {
    it('should render', () => {
      // arrange
      const store = {frontPage: {isActive: false}}

      // act
      const wrapper = shallowWithStore(store, <FrontPage />)

      // assert
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })
})