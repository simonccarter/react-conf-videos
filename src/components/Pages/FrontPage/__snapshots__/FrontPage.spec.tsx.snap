// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontPage FrontPage should render 1`] = `
<pure(Component)
  dispatch={[Function]}
  isActive={false}
/>
`;

exports[`FrontPage FrontPageInner should render 1`] = `
<main
  className="frontPage isActive"
>
  <div
    className="header"
  >
    <div
      className="text isActive"
    >
      <h1
        className="headerTitle"
      >
         React.js Videos
      </h1>
      <p
        className="headerText"
      >
         Search React.js conference videos. 
      </p>
      <p
        className="headerText contribute"
      >
         Contribute 
        <a
          className="contributeLink"
          href="https://github.com/simonccarter/react-conf-videos"
        >
          here
        </a>
        .
      </p>
    </div>
  </div>
  <Connect(pure(Component)) />
</main>
`;
